custom-update-agent/Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM --platform=$BUILDPLATFORM golang:alpine AS builder
+RUN apk update && apk add --no-cache git
+RUN apk update && apk add --no-cache ca-certificates
+WORKDIR /build
+
+COPY . .
+RUN go mod download
+
+ARG TARGETOS TARGETARCH
+RUN GOOS=$TARGETOS GOARCH=$TARGETARCH go build -o app .
+
+FROM scratch
+COPY --from=builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
```

Comment:
This will only work for Debians and may cause confusion if someone try this on a different linux OS. Ref: https://go.dev/src/crypto/x509/root_linux.go

Instead of this we can either skip security verification from the code (which is sort of ok for example apps), we can use container mounts, or can just add description here in the comment what to change. @dimitar-dimitrow wdyt?

Reply:
In the current implementation the default go http Client is used. Do you know whether it performs any security verification? My preferences is for a mount, this way the user can choose to provide a non system CA. However this mounting should be described in a README which is currently missing.

Reply:
As far as I remember (but don't quote me on this) it tries to verify the server certificate whether it is trusted or not. And for this it uses what is available [here](https://go.dev/src/crypto/x509/root_linux.go) as a certificates. I guess this is why @gabriela-yoncheva added this `COPY` line there, because with the alpine base image there is nothing (or almost nothing).

Reply:
Adding a mount point for this would complicate a bit the deployment of this application. The README should include two different `run` parts, one for building and starting as a standard application and second part for containerisation of the app.

Another possible solution for this problem might be to use different base platform layer.